tests/baselines/reference/isomorphicMappedTypeInference.types

```diff
@@ -427,7 +427,7 @@ function f4() {
 
 function makeRecord<T, K extends string>(obj: { [P in K]: T }) {
 >makeRecord : <T, K extends string>(obj: { [P in K]: T; }) => { [P in K]: T; }
->           : ^ ^^ ^^^^^^^^^^^^^^^^^^^^^^                ^^^^^^^^ ^^^^^^^^^^^^
+>           : ^ ^^ ^^^^^^^^^^^^^^^^^^^^^^                ^^^^^^^^^^^^^^^^^^^^^

```

Comment:
Here too.

Reply:
Previously the argument mapped type and return mapped type shared a scope, so when we generated the `P` in the argument, we'd see the _same_ type `P` in the return position and reuse it. Now, we exit the scope and discard the name, and generate a new one from the type for the return mapped type scope.

Reply:
(Do note: the return here is inferred - there are no direct input nodes)

Reply:
Hm, alrighty.